ext/ftp/ftp.c

```diff
@@ -1791,9 +1791,9 @@ static void ftp_ssl_shutdown(ftpbuf_t *ftp, php_socket_t fd, SSL *ssl_handle) {
 					break;
 				default:
 					if ((sslerror = ERR_get_error())) {
-					    ERR_error_string_n(sslerror, buf, sizeof(buf));
+						ERR_error_string_n(sslerror, buf, sizeof(buf));
+						php_error_docref(NULL, E_WARNING, "SSL_read on shutdown: %s", buf);
```

Comment:
This seems wrong. `SSL_get_error` may also return `SSL_ERROR_SYSCALL` in which case OpenSSL indicates that the underlying syscall has failed. This changes hides this error case from the user.

Reply:
Still better that current broken implementation.

Please understand that we have released 7.1.24/7.2.12 with a regression, for which we need an urgent fix.